### DISCUSSION TITLE
List of users who have the admin access on repositories

### DISCUSSION THREAD

**Discussion by KunalDange-2512** [community/community#88555] [Actions]
### Select Topic Area

General

### Body

Hi team,

Can you let us know how to get the list of users who have the admin access on repositories of my ORG.

**Comment by Shahnawazk7944** [comment]
Go to your organization's settings page -> Click "People" in the sidebar -> Filter the list by "Owner" or "Admin" roles.

**Reply by KunalDange-2512** [reply]
I want to fetch the data using script. Any suggestion for that?
Also who has this access to make the changes in settings in github?

**Reply by Shahnawazk7944** [reply]
What data do you want to retrieve? Are you looking to access the list of users in the organization? And of course, the admin has all the necessary access to make changes in the repositories.

**Reply by KunalDange-2512** [reply]
I want to retrieve all the repositories of my org and which users has admin access to which repos.

**Reply by Shahnawazk7944** [reply]
You can retrieve all that using an API call like GraphQL API or REST API.

**Reply by KunalDange-2512** [reply]
Can you give any reference link, it will be helpful.

**Reply by Shahnawazk7944** [reply]
For which GraphQL or REST?

**Reply by KunalDange-2512** [reply]
RESTAPI

**Reply by Shahnawazk7944** [reply]
List all repositories
```GET https://api.github.com/orgs/your-org-name/repos```

For each repository, list the collaborators and their permissions
```GET https://api.github.com/repos/your-org-name/repo-name/collaborators```

Filter the collaborators with "admin" permission with your programming language's capabilities to filter the data.

**Reply by Shahnawazk7944** [reply]
And remember 
You need to authenticate your API requests using an OAuth token. you can generate a token from the GitHub settings page. this approaches can be implemented using your preferred programming language and GitHub API client libraries

**Reply by Shahnawazk7944** [reply]
You can use these references

https://docs.github.com/rest
https://docs.github.com/rest/collaborators/collaborators

https://stackoverflow.com/questions/49738639/github-permissions
https://github.com/orgs/community/discussions/22529
https://stackoverflow.com/questions/71271079/how-can-i-use-github-api-to-get-a-list-of-users-added-to-a-repository

**Reply by Magdenko3777** [reply]
- - [ ] [****](https://itunes.apple.com/app/id1093774792?action=write-review)

**Reply by KunalDange-2512** [reply]
Thanks. I will check and update you.

**Comment by Shahnawazk7944** [comment]


**Reply by Magdenko3777** [reply]
> I can guide for REST for GhrapQL i don't have much knowledge to assist you
